src/main/java/com/google/firebase/auth/FirebaseTokenVerifierImpl.java

```diff
@@ -278,6 +283,18 @@ private boolean containsLegacyUidField(IdToken.Payload payload) {
     return false;
   }
 
+  private void checkTenantId(final FirebaseToken firebaseToken) throws FirebaseAuthException {
+    String tokenTenantId = firebaseToken.getTenantId();
+    if (!Strings.nullToEmpty(tokenTenantId).equals(Strings.nullToEmpty(tenantId))) {
```

Comment:
Following might be a bit more readable:

```
if (!Strings.isNullOrEmpty(this.tenantId)) {
  if (!this.tenantId.equals(tokenTenantId) {
    // ...
  }
}
```

Reply:
That's more readable, yes, but it also behaves differently.

Don't we want to throw an exception if `this.tenantId` is null and `tokenTenantId` is non-empty?

Reply:
Ah I see your point. What if we do this in the constructor then:

```
this.tenantId = Strings.nullToEmpty(tenantId);
```

Then during verification you can do `!this.tenantId.equals(tokenTenantId)`.

Reply:
Moving `Strings.nullToEmpty` to the constructor is a good idea.

But I think we'd need to do `!this.tenantId.equals(Strings.nullToEmpty(tokenTenantId)))` or else we'd say there was a mismatch if both were initially null.

Reply:
I pulled the other `Strings.nullToEmpty` to the previous line where tokenTenantId is defined. I think this is better now.